process/factory/containers/interceptorsContainer.go

```diff
@@ -86,6 +86,30 @@ func (ic *interceptorsContainer) Len() int {
 	return ic.objects.Len()
 }
 
+// Iterate will call the provided handler for each and every key-value pair
+func (ic *interceptorsContainer) Iterate(handler func(key string, interceptor process.Interceptor) bool) {
+	if handler == nil {
+		return
+	}
+
+	for keyVal := range ic.objects.Iter() {
+		key, ok := keyVal.Key.(string)
+		if !ok {
+			continue
```

Comment:
This was an original comment on my pr: maybe we should consider removing invalid data from the cache before continuing (valid on both continue instructions)

Reply:
done.